invoice/src/main/java/org/killbill/billing/invoice/optimizer/InvoiceOptimizerExp.java

```diff
@@ -179,14 +179,7 @@ public boolean apply(final InvoiceItem item) {
                                     }
                                 }).orNull();
                                 if (existingItem != null) {
-                                    if (existingItem.getEndDate().compareTo(invoiceItem.getEndDate()) != 0 ||
```

Comment:
Removing this might not work with pro-ration use-cases? This also comment above.

Reply:
I have a [test](https://github.com/killbill/killbill/blob/invoice-optimization-fix/beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithInvoiceOptimization.java#L334) that creates a trailing pro-ration (early cancelation), and it seems to work well. If you have other scenario in mind, send them over so I can try them out.